octodns/record/__init__.py

```diff
@@ -104,17 +105,23 @@ def register_type(cls, _class, _type=None):
 
     @classmethod
     def new(cls, zone, name, data, source=None, lenient=False):
-        name = str(name).lower()
+        reasons = []
+        try:
+            name = idna_encode(str(name))
+        except IdnaError as e:
```

Comment:
There is a `strict` parameter to the `encode` method of the lib, but unfortunately it doesn't actually disable any checking, it just has a very specific meaning. We need to catch any problems it turns out (as wrapped by our helper exception) and turn them into reasons here. I may someday look at PR'ing a softer mode to the `idna` lib, but I'm guessing that'd be a fairly major (api) change and not likely to land anytime soon.